### DISCUSSION TITLE
Where to start for implementing FSR 2.2.2 in a Vulkan game through a Mod

### DISCUSSION THREAD

**Discussion by CacodemonOfDoom** [godotengine/godot-proposals#9006] [3D]
I'm very new to this topic, so I hope to not say anything redundantly stupid. As I play a lot Doom Eternal and I can't afford a better GPU I already tried using an already existing Mod that implements FSR 2.1 but it still has issues like shimmering and blurring particles or models. before FSR 3.0 will be supported for Vulkan too, i would like to learn how to implement the previous version to this game but I have no idea where to begin with, what pre-requisites are need nor how to start with the project. I do have some experience with C and I'm currently reading a book which clumsily goes over pointers, File I/O and the base functions, and later I will read deeper about pointers and memory allocation if needed. Maybe, if not surely, learn more about C from a broad project like this would be exhausting and tedious, but I dealt with frustrating activities a lot in my life due to my AuDHD, so the will to learn is there, I just don't know where and how to start the whole process and I would really appreciate a guide line about the topic! Thanks for anyone who read this post!

**Comment by Calinou** [comment]
Godot 4.2 already has built-in support for FSR 2.2. [FSR3 is planned to be implemented in the future](https://github.com/godotengine/godot-proposals/issues/8768) once it has a Vulkan port and further stabilizes.

Porting FSR to other engines is off-topic for the Godot proposals repository.
